tests/cartesian_tests/integration_tests/multi_feature_tests/test_code_generation.py

```diff
@@ -586,3 +586,79 @@ def test(out: Field[np.float64], inp: Field[np.float64]):
         backend=backend, aligned_index=(0, 0, 0), shape=(2, 2, 2), dtype=np.float64
     )
     test(out, inp)
+
+
+@pytest.mark.parametrize("backend", ["numpy"])
+def test_K_offset_write(backend):
+    @gtscript.stencil(backend=backend)
+    def simple(A: Field[np.float64], B: Field[np.float64]):
+        with computation(FORWARD), interval(...):
+            B[0, 0, 1] = A
+
+    A = gt_storage.zeros(
+        backend=backend, aligned_index=(0, 0, 0), shape=(2, 2, 4), dtype=np.float64
+    )
+    A[:, :, :] = 42.0
```

Comment:
I didn't look what the quality of the other tests are, but maybe we can fill different values in k, just to make sure we actually see the shifted value, not the original one